test/go.mod

```diff
@@ -0,0 +1,5 @@
+// The test module exists to prevent the integration tests from running when
+// unit tests are run.
+module github.com/hashicorp/consul/test
+
+go 1.14
```

Comment:
I'm not sure I understand the need/ramifications of adding a new nested module even if it's just a placeholder. Nested modules have caused so many unexpected issues already that I'm at least a little wary of adding more but that could well be my ignorance of the real issues so far. Could you elaborate a bit so we can all understand why this is necessary?

Reply:
Could we just use build tags to do this? so `go test`... is just unit tests but you need `go test -tags=envoy_integration` or something?

Reply:
🤔 perhaps that suggestion would also need a filter to avoid running the regular suite too. If we hide it behind the `make` target that seems OK though?

Reply:
Build tags could work. I've used that strategy before and it worked well. We would add an `integration` tag to all the go test files in this directory and the Makefile would set the tag to run it. We won't need negative tags because we can run only tests under `test/integration`.

That approach is slightly nicer for two reasons:
* you can run the test suite from the project root. With the submodule we have to `cd test` first.
* the test code can be linted without adding another line to the `lint` target in the makefile

I suspect this use of a submodule won't hit the problems we've faced so far. The main problem I am aware of is when a package moves from one module to another. When a package exists in two modules (at different versions) you have to be very careful about which versions you use. Since this is a new module we wouldn't have that problem.

That said, I like the idea of using build tags for this. I'll switch to build tags.